providers/implementations/rands/seed_src_jitter.c

```diff
@@ -194,20 +194,20 @@ static int jitter_generate(void *vseed, unsigned char *out, size_t outlen,
         return 0;
     }
 
-    if (adin != NULL && adin_len > 0) {
-        if (!ossl_rand_pool_add(pool, adin, adin_len, 0)) {
-            ERR_raise(ERR_LIB_PROV, ERR_R_RAND_LIB);
-            ossl_rand_pool_free(pool);
-            return 0;
-        }
-    }
-
     /* Get entropy from jitter entropy library. */
     entropy_available = ossl_prov_acquire_entropy_from_jitter(s, pool);
 
     if (entropy_available > 0)
         memcpy(out, ossl_rand_pool_buffer(pool), ossl_rand_pool_length(pool));
 
+    if (adin != NULL && adin_len > 0) {
+        size_t i;
+
+        /* xor the additional data into the output */
+        for (i = 0; i < adin_len; ++i)
+            out[i % outlen] ^= adin[i];
```

Comment:
I know the seed_src does this the exact same way, but does it make sense to iterate if adin_len is greater than outlen here?  The addition of external input is implementation dependent and optional, and restricting the extra input len to outlen would limit run time if someone provided an absolutely huge amount of external data (i.e. SIZE_MAX bytes or some such).

Reply:
I do not think it makes sense to differ from the other places.

Reply:
It would make sense to pull this snippet into a common routine to avoid duplicate code and maintain consistent application of the additional data.

Reply:
> It would make sense to pull this snippet into a common routine to avoid duplicate code and maintain consistent application of the additional data.

That would be for a separate PR

Reply:
I'm not so sure.  Since this is adding the duplicate code, the refactoring could be done here and now.  This avoids another worthwhile but wayward improvement going missing.